### ISSUE TITLE
Unable to use when MacOS tasks are present

### ISSUE THREAD

**Issue by real-or-random** [cirruslabs/cirrus-cli#418]
We have MacOS tasks in our `.cirrus.yml`. These are not supported by `cirrus-cli`, which we are aware of but this is not the issue here.

The problem is that these tasks prevent us from running the Linux tasks:
```
Error: failed to create task "x86_64: macOS Catalina": failed to create instance: got nil instance which means it's probably not supported by the CLI
```

My second try was `run "x86_64: Linux (Debian stable)"` but this doesn't work because the task depends on a "Prebuild ..." task:
```
🕛 'x86_64: Linux (Debian stable)' task (ASM:no BENCH:yes BUILD:check CC:gcc CTIMETEST:yes ECDH:no ECMULTGENPRECISION:auto EXPERIMENTAL:no RECOVERY:yes SCHNORRSIG:no STATICPRECOMPUTATION:yes s
   ✅ Preparing execution environment... 2.3s
   ❌ image pull 1.0s
      Pulling image gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest...
      Failed to pull gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest: Error response from daemon: manifest for gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:l.
      
Error: Error response from daemon: manifest for gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest not found: manifest unknown: Failed to fetch "latest" from request "/v2/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e/manifests/latest".
```

While it's possible to run `cirrus run "Prebuild ci/linux-debian.Dockerfile"`, it's unfortunately not possible to specify multiple tasks, so no luck here. The only thing that works is to edit `.cirrus.yml` and remove the MacOS tasks.

See https://github.com/bitcoin-core/secp256k1/blob/master/.cirrus.yml

Solving #31 would solve this issue. Even just skipping MacOS tasks would solve this issue.

**Comment by fkorotkov** [comment]
After running `cirrus run "Prebuild ci/linux-debian.Dockerfile"` you should get the image locally built and subsequential invocation of `cirrus run --container-lazy-pull "x86_64: Linux (Debian stable)` should work theoretically. 🤔 

Anyways, @edigaryev what do you think of skipping unsupported tasks instead of failing them? Also maybe we should consider adding  `--with-dependencies` flag to `run`. I think we discussed it at some point but don't remember what the conclusion was. 😅

**Comment by edigaryev** [comment]
Hello! Thanks for reporting this.

Sorry, #31 is a bit poorly named: it aims to replicate the cloud's behavior where we try to run all non-conflicting tasks even through some tasks have failed. The build still have to be considered failed, otherwise it's possible to miss important QA-related details. Yet, as far as I understand, you're looking for a clean return code from `cirrus run`.

Unsupported instances are now skipped by default as a result of #420.

The changes should be available shortly in the [0.55.0 release](https://github.com/cirruslabs/cirrus-cli/releases/tag/v0.55.0).

**Comment by jonasnick** [comment]
Thank you for the quick response. Can confirm that 0.55 works just as expected.
